### PR TITLE
78685 write integration tests vol1

### PR DESCRIPTION
**Pull request by llykkelarsen**
DevOps#78685

Positive tests for _most_ endpoints in InvitationsController. 
The PBI was split up due to sprint planning, so any 'missing' tests will come in Vol 2.

### CODE REVIEW

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs [hunk 1]
```diff
@@ -52,52 +56,82 @@ public async Task GetInvitation_AsViewer_ShouldReturnNotFound_WhenUnknownId()
...
                 Guid.NewGuid().ToString(),
                 DisciplineType.DP,
+                _invitationStartTime,
+                _invitationEndTime,
+                _participants,
+                _mcPkgScope,
+                null,
                 HttpStatusCode.Forbidden);
 
         #endregion
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs:137]: Will tests for edit (below) come in Vol2? Add a Todo

Comment (llykkelarsen) [reply src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs:137]: Yes they will 👍  i'll add todo

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs [hunk 2]
```diff
@@ -11,27 +14,249 @@ public async Task GetInvitation_AsViewer_ShouldGetInvitation()
...
+                _mcPkgScope,
+                null
+            );
+
+            var invitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                id);
+
+            // Assert
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs:70]: move Assert-comment above line 65. Add Asserts for Title, Description and Location (if not, remove Title, Description and Location)

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs [hunk 3]
```diff
@@ -11,27 +14,249 @@ public async Task GetInvitation_AsViewer_ShouldGetInvitation()
...
+                TestFactory.PlantWithAccess,
+                id,
+                editInvitationDto);
+
+            var updatedInvitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                id);
+
+            // Assert
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs:129]: move comment up

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs [hunk 4]
```diff
@@ -11,27 +14,249 @@ public async Task GetInvitation_AsViewer_ShouldGetInvitation()
...
+            AssertRowVersionChange(currentRowVersion, newRowVersion);
+            Assert.AreEqual(UpdatedTitle, updatedInvitation.Title);
+            Assert.AreEqual(UpdatedDescription, updatedInvitation.Description);
+        }
+
+        [TestMethod]
+        public async Task UploadAttachment_AsPlanner_ShouldUploadAttachment()
+        {
+            // Arrange
+            var invitationIdForAttachment = await InvitationsControllerTestsHelper.CreateInvitationAsync(
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs:139]: Fine for me, but in this context you could have used existing Invitation found by InitialInvitationId for "attachment container" ... no need for creating a new invit.

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/ParticipantDto.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+
+namespace Equinor.ProCoSys.IPO.WebApi.IntegrationTests.Invitations
+{
+    public class ParticipantDtoGet
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/ParticipantDto.cs:5]: splitt ParticipantDtoGet  and ParticipantDtoEdit into 2 sep files. Same name on file as class

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/KnownTestData.cs [hunk 6]
```diff
@@ -8,10 +8,16 @@ public class KnownTestData
         public static string Plant => "PCS$PLANT1";
         public static string ProjectName => "TestProject";
         public static string ProjectDescription => "Test - Project";
-        public static string Invitation => "TestInvitation";
+        public static string McPkgNo => "MC10-2034";
+        public static string CommPkgNo => "COMM12-2387";
+        public static string DisciplineCode => "A";
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/KnownTestData.cs:13]: remove

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/appsettings.json [hunk 7]
```diff
@@ -18,5 +18,23 @@
   "Synchronization": {
     "UserOid": "00000000-0000-0000-0000-000000000000",
     "Interval": "24:0:0"
+  },
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/appsettings.json:21]: do we need this? Is it used in int tests?

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/appsettings.json:21]: I tried without and the attachment tests fails ... hm .. I don't understand why IPO need this and preservation not. We can have a quick chat about it, but not use to much time

**Review by janid1967 — CHANGES_REQUESTED** [review]
See comments

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/appsettings.json (hunk 7 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/appsettings.json:21]: I see why IPO need it. OK here. However, I don't understand why preservation to not need the Synchronization block in IntegrationTests/appsettings.json. I resolve this comment

**Review by janid1967 — APPROVED** [review]
I approve with suggested changes

**Pull request merged**